### PR TITLE
Update documentation for how 5xxs map to TCP traffic

### PR DESCRIPTION
**Pull request by chusteven**
Updating documentation to make clearer this issue: https://github.com/envoyproxy/envoy/issues/3058

Risk Level: Low since it's just documentation?
Testing: Didn't do any -- happy to but was too lazy to set up my environment 😅 
Docs Changes: Specifically for [outlier detection](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier), I was confused that 5xx mapped -- for TCP traffic -- to connection failures
Release Notes: Updated documentation on outlier detection
Platform Specific Features: N/A
[Optional Runtime guard:] ?
[Optional Fixes #Issue] https://github.com/envoyproxy/envoy/issues/3058
[Optional Fixes commit #PR or SHA] N/A
[Optional Deprecated:] N/A
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):] N/A

### CODE REVIEW

File: api/envoy/api/v2/cluster/outlier_detection.proto [hunk 1]
```diff
@@ -24,9 +24,9 @@ option (udpa.annotations.file_status).package_version_status = FROZEN;
 // more information on outlier detection.
 // [#next-free-field: 21]
 message OutlierDetection {
-  // The number of consecutive 5xx responses or local origin errors that are mapped
-  // to 5xx error codes before a consecutive 5xx ejection
-  // occurs. Defaults to 5.
+  // The number of consecutive 5xx responses or local origin errors (e.g. for TCP traffic,
```
Comment (htuch) [inline api/envoy/api/v2/cluster/outlier_detection.proto:27]: I think the doc callouts in RST are fine, but the API comment changes are misleading, as they over-index on the TCP case.

Comment (chusteven) [reply api/envoy/api/v2/cluster/outlier_detection.proto:27]: That makes sense to me. But from my perspective, part of the issue in the first place was that this was named `5xx` which I immediately assumed meant HTTP traffic only. Is there other wording that we can compromise on?

Maybe:
```
The number of consecutive server-side error responses (for HTTP traffic,
5xx responses; for TCP traffic, connection failures; for Redis, failure to respond
PONG, etc.) before a consecutive 5xx ejection occurs. Defaults to 5.
```

I think also L3 traffic is supported but I don't have any experience with that to know the correct wording to use for when that fails 😓

Comment (htuch) [reply api/envoy/api/v2/cluster/outlier_detection.proto:27]: Let's go with that.

**Comment by adisuissa** [comment]
Please fix DCO.
/wait

**Review by yanavlasov — APPROVED** [review]

**Comment by yanavlasov** [comment]
@adisuissa for API stamp

**Review by adisuissa — APPROVED** [review]
/lgtm api

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**